pandas/core/indexes/datetimelike.py

```diff
@@ -665,10 +665,38 @@ def _time_shift(self, periods, freq=None):
         return type(self)(result, name=self.name)
 
     @deprecate_kwarg(old_arg_name='n', new_arg_name='periods')
-    @Appender(DatetimeLikeArrayMixin.shift.__doc__)
     def shift(self, periods, freq=None):
-        new_values = self._data.shift(periods, freq=freq)
```

Comment:
aside from the docstring, there is a real difference here; I'm pretty sure this is supposed to be self._data._time_shift (like it now is in master)

Reply:
That's correct.